services/federator/src/Federator/Monitor/Internal.hs

```diff
@@ -323,57 +320,84 @@ watchedDirs resolve path = do
   pure (dirs0 ++ dirs1)
 
 data FederationSetupError
-  = InvalidCAStore FilePath
+  = InvalidCAStore FilePath String
   | InvalidClientCertificate String
+  | InvalidClientPrivateKey String
+  | CertificateAndPrivateKeyDoNotMatch FilePath FilePath
+  | SSLException SSL.SomeSSLException
   deriving (Show)
 
 instance Exception FederationSetupError
 
 showFederationSetupError :: FederationSetupError -> Text
-showFederationSetupError (InvalidCAStore path) = "invalid CA store: " <> Text.pack path
+showFederationSetupError (InvalidCAStore path msg) = "invalid CA store: " <> Text.pack path <> ", error: " <> Text.pack msg
 showFederationSetupError (InvalidClientCertificate msg) = Text.pack msg
+showFederationSetupError (InvalidClientPrivateKey msg) = Text.pack msg
+showFederationSetupError (CertificateAndPrivateKeyDoNotMatch cert key) = Text.pack $ "Certificate and private key do not match, certificate: " <> cert <> ", private key: " <> key
```

Comment:
Isn't it a bit dangerous to print the private key here?

Reply:
Its just the path the file.